File: examples/completions_stream.py
import os
import sys

# VoltGate Python Client Integration Example
# To run this script:
#   1. Install dependencies: pip install anthropic dotenv
#   2. Ensure VoltGate is running on localhost:3001
#   3. Run the script: python completions_stream.py

try:
    import anthropic
    from dotenv import load_dotenv
except ImportError:
    print("❌ Error: Missing required dependencies.")
    print("Please install them first using: pip install anthropic python-dotenv")
    sys.exit(1)

# Load variables from .env file
load_dotenv()

# Retrieve API key or use a fallback. VoltGate secures endpoints with ROUTER_API_KEY.
# If ROUTER_API_KEY is not defined in the environment, the router runs in open mode.
router_key = os.getenv("ROUTER_API_KEY", "default-open-key")

print("⚡ Connecting to VoltGate Proxy on http://localhost:3001...")

# Initialize Anthropic Client pointing to VoltGate instead of the native API endpoint
client = anthropic.Anthropic(
    api_key=router_key,
    base_url="http://localhost:3001",
)

# VoltGate automatically routes requests based on task complexity.
# You can request standard models (e.g. claude-sonnet-4-6), and VoltGate will
# evaluate prompt metrics to select the best cost-efficient model.
prompt = "Write a python function to compute the edit distance between two strings with dynamic programming."

print(f"\n💬 Sending Prompt:\n\"{prompt}\"")
print("\n🤖 VoltGate Streaming Response:")
print("-" * 50)

try:
    stream = client.messages.create(
        model="claude-sonnet-4-6",  # Will be dynamically classified & routed
        max_tokens=1024,
        messages=[{"role": "user", "content": prompt}],
        stream=True,
    )

    for event in stream:
        if event.type == "content_block_delta":
            print(event.delta.text, end="", flush=True)
    print()
    print("-" * 50)
    print("\n✓ Stream completed successfully!")
except Exception as e:
    print(f"\n❌ Error contacting VoltGate: {e}")
    print("Is the VoltGate server running on port 3001?")

File: examples/completions_stream.js
// VoltGate Node.js/JavaScript Integration Example
// To run this script:
//   1. Install dependencies: npm install @anthropic-ai/sdk dotenv
//   2. Ensure VoltGate is running on localhost:3001
//   3. Run the script: node completions_stream.js

const path = require('path');

try {
    require('dotenv').config({ path: path.join(__dirname, '../.env') });
} catch (e) {
    // Ignore error if dotenv is missing
}

let Anthropic;
try {
    Anthropic = require('@anthropic-ai/sdk').default;
} catch (e) {
    console.error("❌ Error: Missing required dependency '@anthropic-ai/sdk'.");
    console.error("Please install it first using: npm install @anthropic-ai/sdk dotenv");
    process.exit(1);
}

// Retrieve API key or use a fallback. VoltGate secures endpoints with ROUTER_API_KEY.
// If ROUTER_API_KEY is not defined in the environment, the router runs in open mode.
const routerKey = process.env.ROUTER_API_KEY || "default-open-key";

console.log("⚡ Connecting to VoltGate Proxy on http://localhost:3001...");

const client = new Anthropic({
    apiKey: routerKey,
    baseURL: "http://localhost:3001",
});

const prompt = "Explain the difference between SQL and NoSQL databases in 3 short bullet points.";

console.log(`\n💬 Sending Prompt:\n"${prompt}"`);
console.log("\n🤖 VoltGate Streaming Response:");
console.log("-".repeat(50));

async function main() {
    try {
        const stream = await client.messages.create({
            model: "claude-sonnet-4-6", // Automatically routed based on task complexity
            max_tokens: 1024,
            messages: [{ role: "user", content: prompt }],
            stream: true,
        });

        for await (const chunk of stream) {
            if (chunk.type === 'content_block_delta') {
                process.stdout.write(chunk.delta.text);
            }
        }
        console.log();
        console.log("-".repeat(50));
        console.log("\n✓ Stream completed successfully!");
    } catch (error) {
        console.error(`\n❌ Error contacting VoltGate: ${error.message}`);
        console.error("Is the VoltGate server running on port 3001?");
    }
}

main();
